src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -165,4 +186,10 @@ private async Task<BalancingAuthority> GetBalancingAuthority(Location location,
 
         return balancingAuthority;
     }
+
+    private DateTimeOffset TimeToLowerest(DateTimeOffset date, int minutes = 5)
+    {
+        var d = TimeSpan.FromMinutes(minutes);
+        return new DateTimeOffset(((date.Ticks + d.Ticks - 1) / d.Ticks) * d.Ticks, date.Offset);
```

Comment:
```suggestion
        return new DateTimeOffset((date.Ticks / d.Ticks) * d.Ticks, date.Offset);
```

I think for this to do the right thing the math needs to slightly different than what we do in the mocker.